File: Easy/protocollo/meta_protocollosegview/meta_protocollosegview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_protocollosegview
{
    public class Meta_protocollosegview : Meta_easydata 
	{
        public Meta_protocollosegview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "protocollosegview") {
				Name = "Registrazioni di protocollo";
			EditTypes.Add("seg");
			ListingTypes.Add("seg");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"protanno","protnumero"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "seg": {
						DescribeAColumn(T, "protnumero", "Numero di protocollo", nPos++);
						DescribeAColumn(T, "protanno", "Anno di protocollo", nPos++);
						DescribeAColumn(T, "protocollo_protdata", "Data di protocollo", nPos++);
						DescribeAColumn(T, "protocollo_codiceregistro", "Codice Registro (univoco nell'Istituto)", nPos++);
						DescribeAColumn(T, "protocollo_codiceammipa", "Codice IPA dell'Istituto", nPos++);
						DescribeAColumn(T, "aoo_title", "Area organizzativa omogenea", nPos++);
						DescribeAColumn(T, "registryorigine_title", "Mittente", nPos++);
						DescribeAColumn(T, "protocollo_originemail", "E-mail mittente", nPos++);
						DescribeAColumn(T, "protocollo_originecodiceaoo", "Amministrazione pubblica mittente - Codice IPA area organizzativa omogenea", nPos++);
						DescribeAColumn(T, "protocollo_origineidamm", "Amministrazione pubblica mittente - Codice IPA", nPos++);
						DescribeAColumn(T, "protocollo_oggetto", "Oggetto del documento", nPos++);
						DescribeAColumn(T, "protocollo_annullato", "Annullato", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/protocollo/meta_protocollosegview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_protocollosegview() {
        MetaData.apply(this, ["protocollosegview"]);
        this.name = 'meta_protocollosegview';
    }

    meta_protocollosegview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_protocollosegview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'seg':
						this.describeAColumn(table, 'protnumero', 'Numero di protocollo', null, 1000, null);
						this.describeAColumn(table, 'protanno', 'Anno di protocollo', null, 2000, null);
						this.describeAColumn(table, 'protocollokind_title', 'Ingresso - Uscita - Interno', null, 3200, 255);
						this.describeAColumn(table, 'protocollo_protdata', 'Data di protocollo', null, 4000, null);
						this.describeAColumn(table, 'protocollo_codiceammipa', 'Codice IPA dell\'Istituto', null, 6000, 50);
						this.describeAColumn(table, 'aoo_title', 'Area organizzativa omogenea', null, 7200, 1024);
						this.describeAColumn(table, 'registryorigine_title', 'Mittente', null, 8300, 101);
						this.describeAColumn(table, 'protocollo_originemail', 'E-mail mittente', null, 9000, 512);
						this.describeAColumn(table, 'protocollo_originecodiceaoo', 'Amministrazione pubblica mittente - Codice IPA area organizzativa omogenea', null, 10000, 50);
						this.describeAColumn(table, 'protocollo_origineidamm', 'Amministrazione pubblica mittente - Codice IPA', null, 11000, 1024);
						this.describeAColumn(table, 'protocollo_oggetto', 'Oggetto del documento', null, 12000, 1024);
						this.describeAColumn(table, 'classificazioneprotocollo_title', 'Sottoclassificazione 1', null, 13200, 1024);
						this.describeAColumn(table, 'classificazioneprotocollo2_title', 'Sottoclassificazione 2', null, 14200, 1024);
						this.describeAColumn(table, 'protocollo_annullato', 'Annullato', null, 20000, null);
						this.describeAColumn(table, 'protocollo_dataannullamento', 'Data di annullamento', 'g', 21000, null);
						this.describeAColumn(table, 'protocollo_motivoann', 'Motivo annullamento', null, 22000, -1);
						this.describeAColumn(table, 'protocollo_protannoregistro', 'Anno di protocollo del registro', null, 27000, null);
						this.describeAColumn(table, 'protocollo_protregistro', 'Numero di protocollo del registro', null, 28000, null);
//$objCalcFieldConfig_seg$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["protanno", "protnumero"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "seg": {
						return "protnumero desc, protanno desc";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('protocollosegview', new meta_protocollosegview('protocollosegview'));

	}());
